app/src/main/java/com/morlunk/mountie/MountieActivity.java

```diff
@@ -79,4 +82,15 @@ public boolean onOptionsItemSelected(MenuItem item) {
 //        }
         return super.onOptionsItemSelected(item);
     }
+
+    public void onClickDetectDevices(View v) {
+        if (mountieService != null)
+        {
+            mountieService.detectDevices();
+        }
+        else
+        {
+            // TODO
```

Comment:
This case isn't necessary to provide, the only non-failure case where the service would be null is very soon after the app's resume.